### PR TITLE
Properly handle on file close events

### PR DESCRIPTION
**Pull request by williamsyang-work**
### What it does

VSCode does not allow developers to have a simple 'onThisFileClose' or 'onAnyFileClose' event. So we had to use a different event to trigger the 'onClose' event handler for customized view configuration.

The one that was being used `onDidChangeVisibleTextEditors` was not working as expected.  The event handler has been changed to `onDidChangeTabs`.

Fixes https://github.com/eclipse-cdt-cloud/vscode-trace-extension/issues/334

### How to test

Open, close, drag and drop files around the IDE with the config editor open.  It should not bug out

### Follow-ups

Continued customized view improvements

### Review checklist

- [ x] As an author, I have thoroughly tested my changes and carefully followed the instructions in this template

### CODE REVIEW

File: vscode-trace-extension/src/json-editor/json-editor.ts [hunk 1]
```diff
@@ -101,14 +101,16 @@ export class JsonConfigEditor {
...
+                viewColumn: vscode.ViewColumn.Beside,
+                preview: false,
             });
             await this.schemaService.setOpenFileSchema(this.tempFileUri, this.schema);
 
             // Return a promise that resolves when the editor is closed
             return new Promise<CustomizationSubmission>((resolve, reject) => {
-                this.closeSubscription = vscode.window.onDidChangeVisibleTextEditors(async editors => {
-                    if (this.trackedEditor && !editors.some(e => e.document.uri.fsPath === this.tempFilePath)) {
+                this.closeSubscription = vscode.window.tabGroups.onDidChangeTabs(async event => {
```
[the remark this answers is not in the document]
Comment (bhufmann) [reply vscode-trace-extension/src/json-editor/json-editor.ts:111]: @williamsyang-work any news on this PR?

[the remark this answers is not in the document]
Comment (bhufmann) [reply vscode-trace-extension/src/json-editor/json-editor.ts:111]: @williamsyang-work what's your plan for this PR? Have you looked into Marc's suggestion about using above?

**Review by bhufmann — CHANGES_REQUESTED** [review]
I tested the PR and the error doesn't happen anymore. I also tested the corner cases when moving the editor in the background. 

I noticed another bug that when the Views view is disposed (e.g. by going to the file explorer), submitting to the back-end doesn't happened while the FE is showing successful submission. This is because the Views view was disposed before receiving the config spec reply. This is out of scope of this PR, just an FYI for a follow-up.

Please fix also the linter errors by running `yarn format:write`

**Review by bhufmann — APPROVED** [review]
Looks good to me. Thanks!
